makefile

```diff
@@ -1,3 +1,53 @@
-proto:
-	@mkdir -p gopb
-	protoc --go_out=plugins=grpc:gopb *.proto
+# start project configuration
+buildDir := build
+pbDir := gopb
+# end project configuration
+
+# start environment setup
+gobin := go
+ifneq (,$(GOROOT))
+gobin := $(GOROOT)/bin/go
+endif
+
+goCache := $(GOCACHE)
+ifeq (,$(goCache))
+goCache := $(abspath $(buildDir)/.cache)
+endif
+goModCache := $(GOMODCACHE)
+ifeq (,$(goModCache))
+goModCache := $(abspath $(buildDir)/.mod-cache)
+endif
+
+ifneq ($(goCache),$(GOCACHE))
+export GOCACHE := $(goCache)
+endif
+ifneq ($(goModCache),$(GOMODCACHE))
+export GOMODCACHE := $(goModCache)
+endif
```

Comment:
Is all of this other environment setup necessary? Seems like the only things you need are the proto-related things and not GOROOT/GOCACHE/GOMODCACHE.

Reply:
Well we are using `go install` so I figured the go setup would be necessary.